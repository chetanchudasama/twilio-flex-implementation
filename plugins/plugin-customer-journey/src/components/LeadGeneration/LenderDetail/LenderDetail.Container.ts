import { connect } from "react-redux";
import { Dispatch } from "redux";
import { CustomerDetailModel } from "@common/components";
import { LenderDetail } from "./LenderDetail";
import { AppState, namespace } from "../../../states";
import { StateToProps, DispatchToProps } from "./LenderDetail.Props";

const mapStateToProps = (state: AppState): StateToProps => {
  const customer: CustomerDetailModel = Object.assign(
    new CustomerDetailModel(),
    state[namespace].customer.customer
  );

  return {
    lenderName: customer.preferredLender?.lenderName ?? "",
  };
};

const mapDispatchToProps = (dispatch: Dispatch<any>): DispatchToProps => ({});

export default connect(mapStateToProps, mapDispatchToProps)(LenderDetail);
