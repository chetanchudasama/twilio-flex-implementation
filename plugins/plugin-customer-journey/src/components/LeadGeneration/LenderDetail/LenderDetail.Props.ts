export interface StateToProps {
  lenderName: string;
}

export interface DispatchToProps {}

export type LenderDetailProps = StateToProps & DispatchToProps;
