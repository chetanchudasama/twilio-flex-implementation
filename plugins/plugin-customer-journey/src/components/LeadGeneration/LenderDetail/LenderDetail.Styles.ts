import styled from "react-emotion";
import { darkGrey, white, black } from "@common/components";

export const LenderDetailStyles = styled("div")({
  padding: 24,
  background: white,
  marginLeft: -8,
  ".lender-detail-heading": {
    fontSize: 20,
    lineHeight: "24px",
    fontWeight: 600,
    color: black,
    marginBottom: 4,
  },
  ".preferred-lender": {
    display: "flex",
  },
  ".preferred-lender-icon": {
    marginRight: 12,
  },
  ".preferred-lender-icon svg": {
    height: 20,
    width: 16,
  },
  ".preferred-lender-name": {
    color: darkGrey,
  },
});
