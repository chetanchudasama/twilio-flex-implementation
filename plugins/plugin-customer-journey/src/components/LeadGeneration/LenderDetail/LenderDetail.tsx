import React from "react";
import { Grid } from "@material-ui/core";
import { secondary, PreferredLenderIcon } from "@common/components";

import { LenderDetailStyles } from "./LenderDetail.Styles";
import { LenderDetailProps } from "./LenderDetail.Props";

export const LenderDetail: React.FC<LenderDetailProps> = ({ lenderName }) => {
  return (
    <LenderDetailStyles>
      <Grid container spacing={16}>
        <Grid item xs={12} className="lender-detail-heading">
          Lender Details
        </Grid>
        <Grid item xs={12} className="preferred-lender">
          <span className="preferred-lender-icon">
            <PreferredLenderIcon color={secondary} opacity="0.4" />
          </span>
          <span className="preferred-lender-name">
            Preferred Lender: {lenderName}
          </span>
        </Grid>
      </Grid>
    </LenderDetailStyles>
  );
};
