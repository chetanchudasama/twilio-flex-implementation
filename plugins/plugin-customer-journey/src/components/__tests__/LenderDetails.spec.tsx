import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";

import { LenderDetailProps } from "../LeadGeneration/LenderDetail/LenderDetail.Props";
import { LenderDetail } from "../LeadGeneration/LenderDetail/LenderDetail";

const props: LenderDetailProps = {
  lenderName: "123 Debt Solutions",
};

jest.mock("@common/components", () => {
  return {
    __esModule: true,
    PreferredLenderIcon: () => {
      return <div>Preferred Lender Icon</div>;
    },
  };
});

describe("LenderDetail", () => {
  it("renders correctly on load", async () => {
    const wrapper = create(<LenderDetail {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should render lender name correctly", () => {
    const wrapper = mount(<LenderDetail {...props} />);

    expect(wrapper.find(".preferred-lender-name").last().html()).toContain(
      props.lenderName
    );
  });
});
