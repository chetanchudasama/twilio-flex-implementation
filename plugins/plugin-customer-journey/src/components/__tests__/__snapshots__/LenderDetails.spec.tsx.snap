// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LenderDetail renders correctly on load 1`] = `
<div
  className="css-cbti2r"
>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 lender-detail-heading"
    >
      Lender Details
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 preferred-lender"
    >
      <span
        className="preferred-lender-icon"
      >
        <div>
          Preferred Lender Icon
        </div>
      </span>
      <span
        className="preferred-lender-name"
      >
        Preferred Lender: 
        123 Debt Solutions
      </span>
    </div>
  </div>
</div>
`;
